### PR TITLE
Build full mesa

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

File: M/Mesa/build_tarballs.jl [hunk 1]
```diff
@@ -3,41 +3,79 @@
...
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-  Dependency("Zlib_jll")
+  Dependency("Zlib_jll"),
+  Dependency("Zstd_jll"),
+  Dependency("XML2_jll"),
+  Dependency("Xorg_libX11_jll"),
+  Dependency("Xorg_xorgproto_jll"),
```
Comment (giordano) [inline M/Mesa/build_tarballs.jl:64]: These are only header files
```suggestion
  BuildDependency("Xorg_xorgproto_jll"),
```

Comment (vchuravy) [reply M/Mesa/build_tarballs.jl:64]: Oh yeah many of those need to be classify

**Comment by vchuravy** [comment]
> ┌ Warning: Linked library libLLVM-8.so could not be resolved and could not be auto-mapped

Seems to be picking up the system llvm instead of the one installed

Likely an issue that meson uses `llvm-config`, which of course doesn't work yet.

**Comment by barche** [comment]
I never managed to make the LLVM stuff work when I tried to build this the first time, why is it needed?

**Comment by vchuravy** [comment]
To fix the fun things like https://github.com/JuliaLang/julia/issues/37200

Right now you can't use GLMakie or anything touching the mesa stack on machines with an AMDGPU or that need one of the software rasterizers.

**Comment by vchuravy** [comment]
@barche is the current form of Mesa actually used? I can of course add back the windows binaries, but it almost makses sense to have two different `jll`s

**Comment by vchuravy** [comment]
Ok so this should work on Linux(gnu) and Windows, I don't see a use-case for Mac yet and I don't know if Mesa is used by FreeBSD.

I am bit hesitant on merging this it has an implicit dependency on LLVM, and thus the Julia version. @staticfloat this would be a prime test-candidate for the tags system I keep talking about.

**Comment by barche** [comment]
Sorry, lost track of this. The current Mesa was only there as a stop-gap for Qt's OpenGL support, at runtime it should be replaced with whatever driver is on the system. My implicit idea was that it should probably never be used by anything that needs fast rendering, so the non-LLVM version seemed sufficient to me.

**Comment by vchuravy** [comment]
Yeah... I am trying to fix the real issue in LLVM proper, since I now finally understand the issue...
But I agree we shouldn't get into the business of shipping a full Mesa.

**Pull request closed without merging**